docs/architecture/devops-for-aspnet-developers/actions-vs-pipelines.md

```diff
@@ -0,0 +1,134 @@
+---
+title: DevOps with .NET and GitHub Actions - Compare GitHub Actions with Azure Pipelines
+description: GitHub Actions and Azure Pipelines compared and contrasted for decision makers
+author: colindembovsky
+ms.date: 03/04/2021
+---
+
+# Compare and contrast GitHub Actions and Azure Pipelines
+
+[GitHub Actions](https://docs.github.com/actions) and [Azure Pipelines](/azure/devops/pipelines/get-started/what-is-azure-pipelines?view=azure-devops&preserve-view=true) have a common history. In fact, the Actions agent is a fork of the Pipelines agent. There are many similarities between GitHub Actions and Azure Pipelines and it's worth comparing and contrasting them.
+
+## Pipelines as code
+
+Before you compare GitHub Actions and Azure Pipelines, you should consider the benefits of *pipelines as code*. Pipelines as code:
+
+> [!div class="checklist"]
+
+> * Benefit from standard source control practices (such as code reviews via pull request and versioning).
+> * Can be audited for changes just like any other files in the repository.
+> * Don’t require accessing a separate system or UI to edit.
+> * Can fully codify the build, test, and deploy process for code.
+> * Can usually be templatized to empower teams to create standard processes across multiple repositories.
+
+> [!NOTE]
+> The term "pipelines" can also be referred to by several different interchangeable words: *pipeline*, *workflow*, and *build* are common terms. In this article, references to *Azure Pipelines* are referring to [YAML Pipelines](/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops&preserve-view=true#define-pipelines-using-yaml-syntax), and not the older UI-based [Classic Pipelines](/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops&preserve-view=true#define-pipelines-using-the-classic-interface).
+
+## Agents and runners
+
+Before you examine pipelines themselves, you should consider how these pipelines *execute*. Both GitHub Actions and Azure Pipelines are really *orchestration engines*. When a pipeline is triggered, the system finds an "agent" and tells the agent to execute the jobs defined in the pipeline file.
+
+Azure Pipelines run on *agents*. The agent is written in .NET, so it will run wherever .NET can run: Windows, macOS, and Linux. Agents can even run in containers. Agents are registered to a [pool](/azure/devops/pipelines/agents/pools-queues?view=azure-devops&preserve-view=true) in Azure Pipelines or to a repository or organization in GitHub. Agents can be *hosted* or *private*.
```

Comment:
```suggestion
Azure Pipelines run on *agents*. The agent is written in .NET, so it will run wherever .NET can run: Windows, macOS, and Linux. Agents can even run in containers. Agents are registered to a [pool](/azure/devops/pipelines/agents/pools-queues) in Azure Pipelines or to a repository or organization in GitHub. Agents can be *hosted* or *private*.
```